src/naive/time/u31.rs

```diff
@@ -0,0 +1,287 @@
+use core::{cmp, fmt, hash, mem};
+
+/// A 31-bit unsigned integer
+#[allow(unreachable_pub)] // public through `rkyv::Archive<NaiveTime>`
+#[repr(transparent)]
```

Comment:
I suppose this is necessary in order for the compiler to leverage the niche? What does it mean for semver compatibility going forward?

Reply:
It's not a public type, so the user should not need to care how U31 is implemented. Unless a user does very odd stuff like accessing an `Option<NaiveTime>` in C, they should not be bothered by this change.

Actually, I think I don't need `struct U31(Buf)` anyhow. I could just let U31 be the endianness-guarded struct directly.